src/RA_Dlg_MemBookmark.cpp

```diff
@@ -503,26 +503,26 @@ void Dlg_MemBookmark::PopulateList()
     if (m_vBookmarks.size() == 0)
         return;
 
+    // TBD: Make control handles wrapped in unique_ptr as data members for member functions
     HWND hList = GetDlgItem(m_hMemBookmarkDialog, IDC_RA_LBX_ADDRESSES);
     if (hList == nullptr)
         return;
 
-    int topIndex = ListView_GetTopIndex(hList);
+    _UNUSED int topIndex = ListView_GetTopIndex(hList);
     ListView_DeleteAllItems(hList);
     m_nNumOccupiedRows = 0;
 
-    for (MemBookmark* bookmark : m_vBookmarks)
+    // bookmark isn't used so let's use iterators
+    for (auto it = m_vBookmarks.cbegin(); it != m_vBookmarks.cend(); ++it)
     {
-        LV_ITEM item;
-        ZeroMemory(&item, sizeof(item));
-        item.mask = LVIF_TEXT;
-        item.cchTextMax = 256;
-        item.iItem = m_nNumOccupiedRows;
-        item.iSubItem = 0;
-        item.iItem = ListView_InsertItem(hList, &item);
-
-        ASSERT(item.iItem == m_nNumOccupiedRows);
-
+        // kind of like zero memory
+        auto lpItem{ std::make_unique<LV_ITEM>() }; 
+        lpItem->mask       = LVIF_TEXT;
+        lpItem->iItem      = m_nNumOccupiedRows;
+        lpItem->cchTextMax = 256;        
+
+        lpItem->iItem = ListView_InsertItem(hList, lpItem.get());
+        ASSERT(lpItem->iItem == m_nNumOccupiedRows);
```

Comment:
There was a warning about the enumerable not being used. So we did the same thing just with iterators. (you need a `begin `and `end `to use ranged for).

A ranged for works exactly like `foreach `in C#, there is a VC++ "`for each`" that works and looks exactly like it but it's not portable.